pkg/util/clickhouse/clickhouse.go

```diff
@@ -63,12 +69,19 @@ func Connect(url string) (*sql.DB, error) {
 	if err != nil {
 		return connect, fmt.Errorf("failed to open ClickHouse: %v", err)
 	}
-	if err := connect.Ping(); err != nil {
-		if exception, ok := err.(*clickhouse.Exception); ok {
-			return connect, fmt.Errorf("failed to ping ClickHouse: %v", exception.Message)
-		} else {
-			return connect, fmt.Errorf("failed to ping ClickHouse: %v", err)
+	var errMessages []error
+	if err = wait.PollImmediate(pingRetryInterval, pingTimeout, func() (done bool, err error) {
+		if err := connect.Ping(); err != nil {
+			if exception, ok := err.(*clickhouse.Exception); ok {
+				errMessages = append(errMessages, fmt.Errorf("failed to ping ClickHouse, message: %v", exception.Message))
+			} else {
+				errMessages = append(errMessages, fmt.Errorf("failed to ping ClickHouse, err: %v", err))
```

Comment:
Nit: Looks like the each error message in the `errMessages` list is separated by a space, which could make it slightly difficult for readers to distinguish between error messages as shown in the unit tests:
`error list: [failed to ping ClickHouse, message: error 1 failed to ping ClickHouse, err: error 2"`
Maybe consider removing the repeated message `failed to ping ClickHouse` and adding a custom separator for better readability.

Reply:
Thanks for reviewing. I've updated. Please take a look to see if it is clear now?